### PR TITLE
Transform v2 -> v1 stack, stackFrame, region

### PR DESCRIPTION
**Pull request by EasyRhin0**
Please check the logicalLocations (aka the cold trail from hell), I'm not confident there.

### CODE REVIEW

File: src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs [hunk 1]
```diff
@@ -667,5 +667,231 @@ public void SarifTransformerTests_ToVersionOne_OneRunWithInvocationAndNotificati
...
+                  ""frames"": [
+                    {
+                      ""location"": {
+                        ""physicalLocation"": {
+                          ""fileLocation"": {
+                            ""uri"": ""file:///C:/src/main.cs""
+                          },
+                          ""region"": {
+                            ""startLine"": 15,
+                            ""startColumn"": 9
```
Comment (lgolding) [inline src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs]: @michaelcfanning @EasyRhin0 

Chris, you have no action for now, but be aware:

At TC #&#xFEFF;18 on May 30th, we approved changes to the default values of missing region properties. In particular, if `endColumn` is missing, in v1 that meant `endColumn` == `startColumn` (an insertion point). But in v2, `endColumn` defaults to “one character past EOL (excluding the newline sequence”).

The reason for this change is that one of the TC members argued persuasively that the most natural interpretation of the region `{ "startLine": 3 }` is "The whole of line 3" and not "An insertion point at the beginning of line 3."

So when you get around to making the schema changes for oasis-tcs/sarif-spec#93, you'll need to adjust the transformations as well.

Comment (EasyRhin0) [reply src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs]: OK thanks

File: src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs [hunk 2]
```diff
@@ -667,5 +667,231 @@ public void SarifTransformerTests_ToVersionOne_OneRunWithInvocationAndNotificati
...
+@"{
+  ""$schema"": ""http://json.schemastore.org/sarif-1.0.0"",
+  ""version"": ""1.0.0"",
+  ""runs"": [
+    {
+      ""tool"": {
+        ""name"": ""CodeScanner"",
+        ""semanticVersion"": ""2.1.0""
+      },
+      ""invocation"": {},
```
Comment (lgolding) [inline src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs]: Don't emit empty object.

Comment (EasyRhin0) [reply src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs]: This happens when there are notifications but no other properties set on the invocation.

File: src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs [hunk 3]
```diff
@@ -667,5 +667,231 @@ public void SarifTransformerTests_ToVersionOne_OneRunWithInvocationAndNotificati
...
+                  ""module"": ""RuleLibrary"",
+                  ""threadId"": 52,
+                  ""fullyQualifiedLogicalName"": ""Rules.SecureHashAlgorithmRule.Evaluate"",
+                  ""address"": 10092852
+                },
+                {
+                  ""uri"": ""file:///C:/src/main.cs"",
+                  ""module"": ""RuleLibrary"",
+                  ""threadId"": 52,
+                  ""fullyQualifiedLogicalName"": ""Rules.SecureHashAlgorithmRule.Register-0"",
```
Comment (lgolding) [inline src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs]: In v2, the `stackFrame.fullyQualifiedLogicalName` property is always the key into the files dictionary (if the files dictionary exists). So you have to do this:
```
If the files dictionary exists
    AND it has a key that matches v2 stackFrame.fullyQualifiedLogicalName
    AND the file object specified by that key has a fullyQualifiedName property
THEN
    set v1 stackFrame.fullyQualifiedLogicalName to v2 file.fullyQualifiedName
    set v1 stackFrame.logicalLocationKey to v2 stackFrame.fullyQualifiedLogicalName
ELSE
    set v1 stackFrame.fullyQualifiedLogicalName to v2 stackFrame.fullyQualifiedLogicalName
```

File: src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs [hunk 4]
```diff
@@ -204,6 +206,43 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(PhysicalLocation v2Ph
...
+                                      v2Region.EndColumn,
+                                      v2Region.EndLine,
+                                      v2Region.Length,
+                                      v2Region.Offset);
+            }
+
+            return region;
+        }
+
+        internal RegionVersionOne CreateRegion(int startColumn, int startLine, int endColumn = 0, int endLine = 0, int length = 0, int offset = 0)
```
Comment (lgolding) [inline src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs]: Why do you need this overload?

File: src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs (hunk 2 above)
Comment (EasyRhin0) [reply src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs]: Maybe. I'd have to check all of the strings for null/whitespace, and the processId == 0, and the datetimes == datetime.minvalue. Would that be deterministic?

File: src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs (hunk 4 above)
Comment (EasyRhin0) [reply src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs]: Copied from the other transformer, where I do need it. Not needed here yet, will remove.

File: src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs (hunk 2 above)
Comment (lgolding) [reply src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs]: Yes, but now that you spell it out, not worth the trouble. After all, if we're copying an entire run to the property bag, I've got now reason to gripe about this. Let it go.

Comment (EasyRhin0) [reply src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs]: OK

File: src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs (hunk 3 above)
Comment (EasyRhin0) [reply src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs]: Fixed. This actually clarifies it quite a bit, thanks

**Pull request merged**